app/src/main/java/com/mdg/droiders/samagra/shush/AlarmScheduler.java

```diff
@@ -30,52 +36,299 @@ public AlarmScheduler(Context context) {
+        startTime.set(Calendar.MONTH, rightNow.get(Calendar.MONTH));
+        startTime.set(Calendar.DAY_OF_MONTH, rightNow.get(Calendar.DAY_OF_MONTH));
+        endTime.set(Calendar.YEAR, rightNow.get(Calendar.YEAR));
+        endTime.set(Calendar.MONTH, rightNow.get(Calendar.MONTH));
+        endTime.set(Calendar.DAY_OF_MONTH, rightNow.get(Calendar.DAY_OF_MONTH));
+        Log.d(LOG_TAG + "now", LOG_DATE_FORMAT.format(rightNow.getTime()));
+
+        // If startTime's hour is ahead of endTime's, assume that
+        // endTime is of that of next day
+        if (startTime.get(Calendar.HOUR_OF_DAY)
+                > endTime.get(Calendar.HOUR_OF_DAY)) {
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+        }
+
+        // If endTime is in the past increment both start and end time
+        if (endTime.compareTo(rightNow) < 0) {
+            startTime.add(Calendar.DAY_OF_MONTH, 1);
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+        }
+
+        // get Day index of start Time
+        int i = getDay(startTime);
+        for (int j = i; j < 7; j++) {
+            Log.d(LOG_TAG + "I-Val", String.valueOf(j));
+            if (days[j]) {
+                setAlarm(startTime.getTimeInMillis(),
+                        endTime.getTimeInMillis(),
+                        getStartDayID(j, rowID),
+                        getEndDayID(j, rowID));
+            }
+            // Update startTime and endTime
+            startTime.add(Calendar.DAY_OF_MONTH, 1);
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+        }
+        for (int j = 0; j < i; j++) {
+            startTime.add(Calendar.DAY_OF_MONTH, 1);
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+            Log.d(LOG_TAG + "I-Val", String.valueOf(j));
+            if (days[j]) {
+                setAlarm(startTime.getTimeInMillis(),
+                        endTime.getTimeInMillis(),
+                        getStartDayID(j, rowID),
+                        getEndDayID(j, rowID));
+            }
+        }
+    }
+
+    /**
+     * Sets a weekly alarm to silence your phone for every day.
+     * <br><br>
+     * <b>Uses</b> : {@link #setWeeklyAlarm(Calendar, Calendar, boolean[], Integer)} to set weekly alarms.
+     *
+     * @param startHour    The hour at which phone is to be shushed in 24 hour format
+     * @param startMinutes The minute at which phone is to be shushed
+     * @param endHour      The hour at which phone is to be un-shushed in 24 hour format
+     * @param endMinutes   The minute at which phone is to be un-shushed
+     * @param days         A boolean array that is true if the alarm is to be
+     *                     scheduled for the corresponding day. i.e. days[0] is set to be true
+     *                     then a weekly alarm will be set such that it will shush the phone every Monday.
+     * @param rowID        Unique primary key of the shush alarm row.
+     */
+    public void setWeeklyAlarm(int startHour, int startMinutes, int endHour,
```

Comment:
Reffering to [this](https://softwareengineering.stackexchange.com/a/135358) and [this](https://softwareengineering.stackexchange.com/a/135331) answer to the question [When is method overloading appropriate?](https://softwareengineering.stackexchange.com/q/135313), the answers provided by various people suggest that overloading is appropriate when - 

>  Each overloaded method should be functionally the same as the others in an overloaded group, otherwise it will be unclear as to how, when or why behaviour is being varied.

The second answer suggests that -
> From the information given they appear to be equivalent (since one calls the other) and I would think it reasonable to overload them.

And since the methods I overloaded do exactly the same thing(since one calls the other), I think it can be used here.
What do you think @suyashmahar ?